frontend/pages/review.tsx

```diff
@@ -64,14 +65,23 @@ function ApprovalList({ category, filter }: { category: ApprovalCategory; filter
 
   const onCancel = () => {
     setOpen(false)
+    setShowAlert(false)
+    setErrorMessage('')
   }
 
   const onConfirm = async () => {
-    await postEndpoint(`/api/v1/approval/${approval?._id}/respond`, { choice }).then((res) => res.json())
-
-    mutateApprovals()
-    mutateNumApprovals()
-    setOpen(false)
+    await postEndpoint(`/api/v1/approval/${approval?._id}/respond`, { choice }).then((res) => {
+      if (res.status !== 200) {
+        res.json().then((errorResponse) => {
```

Comment:
Make this an async function, then you can write

```
const errorResponse = await res.json()
setShowAlert(true)
setErrorMessage(errorResponse.message)
```
Which is generally clearer.